aten/src/ATen/native/miopen/RNN_miopen.cpp

```diff
@@ -212,7 +260,25 @@ struct RNNDescriptors {
     TensorDescriptor cy_desc;
 
     RNNDescriptors(const RNNParams& fn, miopenHandle_t handle, Tensor x, Tensor y, Tensor hx, Tensor cx) {
-        rnn_desc = fn.rnn.descriptor();
+        if (fn.rnn.dropout_rate == 0.0) {
+            rnn_desc = fn.rnn.descriptor();
+        } else {
+            size_t statesSizeInBytes = 0;
+            miopenDropoutGetStatesSize(handle, &statesSizeInBytes);
+            size_t states_size = statesSizeInBytes / sizeof(rocrand_state_xorwow);
+
+            dropout_states = std::unique_ptr<Workspace>(new Workspace(states_size * sizeof(rocrand_state_xorwow)));
```

Comment:
Please use C++14 paradigms
```suggestion
            dropout_states = std::make_unique<Workspace>(states_size * sizeof(rocrand_state_xorwow));
```